src/utils/appinsights.ts

```diff
@@ -46,6 +49,7 @@ export function initAppInsights(
   appInsights.defaultClient.addTelemetryProcessor(
     removeQueryParamsPreprocessor
   );
+  appInsights.defaultClient.addTelemetryProcessor(sessionIdPreprocessor);
```

Comment:
consider also using a telemetryinitializer
https://docs.microsoft.com/it-it/azure/azure-monitor/app/api-filtering-sampling#itelemetryprocessor-and-itelemetryinitializer
> Use TelemetryInitializers to enrich telemetry with additional properties, or override existing one. Use TelemetryProcessor to filter out telemetry.

Reply:
I wanted to, but I couldn't find any reference in the nodejs lib, I believe they're implemented only in the .net library

Reply:
I've found this https://docs.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics with this note:

> In Node.js projects, you can use new applicationInsights.TelemetryClient(instrumentationKey?) to create a new instance, but this is recommended only for scenarios that require isolated configuration from the singleton defaultClient.

Reply:
yes @BurnedMarshal , it's also documented [here](https://github.com/Microsoft/ApplicationInsights-node.js#use-multiple-instrumentation-keys) - that is needed when you want to send events to multiple appinsights accounts (not our use case)